packages/jaeger-ui/src/components/Monitor/ServicesView/serviceGraph.test.js

```diff
@@ -105,38 +110,47 @@ describe('<ServiceGraph>', () => {
         },
       ],
     });
-    expect(wrapper).toMatchSnapshot();
+    expect(wrapper.find('Tooltip').exists()).toBe(true);
   });
 
-  it('Crosshair mouse hover test', () => {
+  it('Tooltip hover test', () => {
     wrapper.setProps({
       ...props,
       loading: false,
       metricsData: {
         ...serviceMetrics.service_call_rate,
         metricPoints: [{ x: 1631271783806, y: null }, ...serviceMetrics.service_call_rate.metricPoints],
       },
-      crosshairValues: [
-        {
-          key: 1,
-          ...serviceMetrics.service_call_rate.metricPoints[0],
-        },
-      ],
     });
-    expect(wrapper.find('AreaSeries').render()).toMatchSnapshot();
-    expect(wrapper.find('LineSeries').render()).toMatchSnapshot();
+    expect(wrapper.find('Area').length).toBeGreaterThan(0);
+    expect(wrapper.find('Line').length).toBeGreaterThan(0);
   });
 
-  it('AreaSeries onNearestX, XYPlot onMouseLeave', () => {
+  it('displays Tooltip with custom formatter', () => {
     wrapper.setProps({
       ...props,
       loading: false,
       metricsData: serviceMetrics.service_call_rate,
     });
-    wrapper.find('AreaSeries').at(0).prop('onNearestX')({ x: 1, y: 2 }, { index: 7 });
-    expect(wrapper.state().crosshairValues).toEqual([{ label: 0.95 }]);
-    wrapper.find('XYPlot').prop('onMouseLeave')();
-    expect(wrapper.state().crosshairValues).toEqual([]);
+    const tooltipComponent = wrapper.find('Tooltip');
+    const exampleValue = 42;
+    const exampleEntry = {
+      payload: {
+        x: 1631271783806,
+        y: 123,
+        label: 0.75,
+      },
+    };
+    const formattedTooltipContent = tooltipComponent.prop('formatter')(exampleValue, 'name', exampleEntry);
+
+    const formattedLabel = new Date(exampleEntry.payload.x).toLocaleTimeString();
+    const expectedTooltipContent = [
+      `label: ${exampleEntry.payload.label * 100}`,
+      `x: ${formattedLabel}`,
+      `Y: ${exampleEntry.payload.y}`,
```

Comment:
nit: this is testing production code with production code (akin to `assert f(x) == f(x)`). I would rather have precise strings in the expected dict - makes for a simpler and easier to read test, which is not tautological.